ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -107,19 +109,20 @@
                       token-user
                       subscription-user))
         (acl/verify-ingest-management-permission
-          request-context :update :provider-object provider-id)
+         request-context :update :provider-object provider-id)
         (acl/verify-subscription-management-permission
-          request-context :update :provider-object provider-id)))))
+         request-context :update :provider-object provider-id)))))
 
 (defn ingest-subscription
   "Processes a request to create or update a subscription. Note, this will allow
   unlimited subscriptions to be created by users for themselves. Revisit if this
   Becomes a problem"
-  [provider-id native-id request]
-  (let [{:keys [body content-type headers request-context]} request]
+  [provider-id opt-native-id request]
+  (let [{:keys [body content-type headers request-context]} request
+        native-id (or opt-native-id (.toString (UUID/randomUUID)))]
```

Comment:
Should we check for a collision here?

Reply:
I think it should already be handled in metadata-db like all the other concepts.

Reply:
But, now we can't just fail the ingest because of a collision. @daniel-zamora is correct. We should not only check for collisions, but also handle it properly. The error should not go back to the provider.

Reply:
Not if it's a generated ID though. Generated IDs should have a retry